### ISSUE TITLE
ros setup fails on Ubuntu 18.04

### ISSUE THREAD

**Issue by mister-walter** [roswell/roswell#351]
Note that this is a different problem than that described in the other "ros setup fails on ubuntu 18.04" issue.

## Environment

```
# uname -a
Linux foo 4.15.0-45-generic #48-Ubuntu SMP Tue Jan 29 16:28:13 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
# ros --version
roswell 19.1.10.96(b4b46ac)
# ros config
setup.time=3758544612
sbcl-bin.version=1.4.16
default.lisp=sbcl-bin
```

## How to reproduce the issue

```sh
# On a newly installed ubuntu 18.04.1 LTS system
sudo apt install emacs-nox build-essential zlib1g-dev automake libcurl4-openssl-dev git
git clone https://github.com/roswell/roswell.git
cd roswell
sh bootstrap
./configure
make
sudo make install
ros setup
```

## Expected behavior

Setup should succeed.

## Actual result

```sh
Installing sbcl-bin...
No SBCL version specified. Downloading platform-table.html to see the available versions...
[##########################################################################]100%
Installing sbcl-bin/1.4.16...
Downloading https://github.com/roswell/sbcl_bin/releases/download/1.4.16/sbcl-1.4.16-x86-64-linux-binary.tar.bz2
[##########################################################################]100%
Extracting sbcl-bin-1.4.16-x86-64-linux.tar.bz2 to /home/foo/.roswell/src/sbcl-1.4.16-x86-64-linux/
Building sbcl-bin/1.4.16... Done.
Install Script for sbcl-bin...
Installing Quicklisp...
up to date. stop
Making core for Roswell...
up to date. stop
Unhandled TYPE-ERROR in thread #<SB-THREAD:THREAD "main thread" RUNNING
                                  {10005205B3}>:
  The value
    NIL
  is not of type
    (OR (VECTOR CHARACTER) (VECTOR NIL) BASE-STRING PATHNAME SYNONYM-STREAM
        FILE-STREAM)

  when binding SB-IMPL::DEFAULTS

Backtrace for: #<SB-THREAD:THREAD "main thread" RUNNING {10005205B3}>
0: (MERGE-PATHNAMES "setup.lisp" NIL #<FUNCTION EQUAL>) [external]
1: (MAIN "roswell")
2: (SB-INT:SIMPLE-EVAL-IN-LEXENV (APPLY (QUOTE MAIN) ROSWELL:*ARGV*) #<NULL-LEXENV>)
3: (SB-INT:SIMPLE-EVAL-IN-LEXENV (ROSWELL:QUIT (APPLY (QUOTE MAIN) ROSWELL:*ARGV*)) #<NULL-LEXENV>)
4: (SB-EXT:EVAL-TLF (ROSWELL:QUIT (APPLY (QUOTE MAIN) ROSWELL:*ARGV*)) NIL NIL)
5: ((LABELS SB-FASL::EVAL-FORM :IN SB-INT:LOAD-AS-SOURCE) (ROSWELL:QUIT (APPLY (QUOTE MAIN) ROSWELL:*ARGV*)) NIL)
6: (SB-INT:LOAD-AS-SOURCE #<CONCATENATED-STREAM :STREAMS NIL {10030EBE73}> :VERBOSE NIL :PRINT NIL :CONTEXT "loading")
7: ((FLET SB-FASL::THUNK :IN LOAD))
8: (SB-FASL::CALL-WITH-LOAD-BINDINGS #<CLOSURE (FLET SB-FASL::THUNK :IN LOAD) {7FF6B3C6F53B}> #<CONCATENATED-STREAM :STREAMS NIL {10030EBE73}>)
9: ((FLET SB-FASL::LOAD-STREAM :IN LOAD) #<CONCATENATED-STREAM :STREAMS NIL {10030EBE73}> NIL)
10: (LOAD #<CONCATENATED-STREAM :STREAMS NIL {10030EBE73}> :VERBOSE NIL :PRINT NIL :IF-DOES-NOT-EXIST T :EXTERNAL-FORMAT :DEFAULT)
11: ((FLET ROSWELL::BODY :IN ROSWELL:SCRIPT) #<SB-SYS:FD-STREAM for "file /usr/local/etc/roswell/setup.ros" {10030E7173}>)
12: (ROSWELL:SCRIPT "/usr/local/etc/roswell/setup.ros")
13: (ROSWELL:RUN ((:EVAL "(ros:quicklisp)") (:SCRIPT "/usr/local/etc/roswell/setup.ros") (:QUIT NIL)))
14: (SB-INT:SIMPLE-EVAL-IN-LEXENV (ROSWELL:RUN (QUOTE ((:EVAL "(ros:quicklisp)") (:SCRIPT "/usr/local/etc/roswell/setup.ros") (:QUIT NIL)))) #<NULL-LEXENV>)
15: (EVAL (ROSWELL:RUN (QUOTE ((:EVAL "(ros:quicklisp)") (:SCRIPT "/usr/local/etc/roswell/setup.ros") (:QUIT NIL)))))
16: (SB-IMPL::PROCESS-EVAL/LOAD-OPTIONS ((:EVAL . "(progn #-ros.init(cl:load \"/usr/local/etc/roswell/init.lisp\"))") (:EVAL . "(ros:run '((:eval\"(ros:quicklisp)\")(:script \"/usr/local/etc/roswell/setup.ros\")(:quit ())))")))
17: (SB-IMPL::TOPLEVEL-INIT)
18: ((FLET SB-UNIX::BODY :IN SB-EXT:SAVE-LISP-AND-DIE))
19: ((FLET "WITHOUT-INTERRUPTS-BODY-7" :IN SB-EXT:SAVE-LISP-AND-DIE))
20: ((LABELS SB-IMPL::RESTART-LISP :IN SB-EXT:SAVE-LISP-AND-DIE))

unhandled condition in --disable-debugger mode, quitting
```

Interestingly, ros seems to run mostly OK after this, but I've encountered issues when using it to run the ACL2 theorem prover on top of it. Particularly, there seems to be an issue with performing a garbage collection before saving an image:
```
[performing final GC...fatal error encountered in SBCL pid 12294(tid 0x7ff0b44fab80):
no scavenge function for object 0x1 (widetag 0x1)
```

This GC error does not occur when using the distribution-provided SBCL package (version `SBCL 1.4.5.debian`)

**Comment by snmsts** [comment]
hmm not clear...

would you show me what is printed when

```
ros -v -v setup
```

**Comment by mister-walter** [comment]
I was able to reproduce in a VirtualBox VM using the [Ubuntu alternative server ISO](http://cdimage.ubuntu.com/releases/18.04.1/release/) (not sure if that's relevant but it's what I used).

Basically I just installed, updated all installed packages, rebooted, and ran the series of commands described in my initial post (except I used the -v -v flags).

Here's the output from the command you asked:
[roswell_err.txt](https://github.com/roswell/roswell/files/2857945/roswell_err.txt)

The problem seems to be that sbcl-bin.version is screwed up - in the file above it contains some non-printable ASCII characters.

**Comment by snmsts** [comment]
hmm.

**Comment by tcw** [comment]
Have you tried?

apt-get install libev-dev

**Comment by t-sin** [comment]
I meet this behavior too. It's not trouble for my ordinary uses but it's trouble on building executable binary in docker container...

This is result of `ros -v -v setup`. I hope it helps resolving.

```
$ ros -v -v setup
opt:verbose:1
dispatch:-v,name=top
proc_options:-v:top
opt:verbose:3
s 94784287056400 util-list.c 147 "-v":done
dispatch:setup,name=top
proc_cmd:
s 94784287056160 util-dir.c 81 "ROSWELL_HOME":done
q 94784287056400 94784267749202 util-dir.c 83 ".roswell/"
q 94784287056400 94784287022800 cmd-internal.c 157 "/usr/local/etc/roswell/"
s 94784287053600 proc-opt.c 101 "/home/grey/.roswell/":done
s 94784287056400 proc-opt.c 101 "/usr/local/etc/roswell/":done
dispatch:/usr/local/etc/roswell/setup.ros,name=top
proc_cmd:
cmd_script_frontend:0
frontend:script_*:argc=1 argv[0]=/usr/local/etc/roswell/setup.ros
ros_script_cmd=exec ros -Q -N roswell -- $0 "$@"
s 94784287057008 util-cmdline.c 39 "$@":done
dispatch:-Q,name=top
proc_options:-Q:top
opt:quicklisp:2
s 94784287057008 util-list.c 147 "-Q":done
dispatch:-N,name=top
proc_options:-N:top
get_opt(roswellenv,0)=(null)
take1:roswellenv:roswell,(null)
set_opt(roswellenv)='roswell'
s 94784287059664 util-list.c 147 "-N":done
s 94784287059280 util-list.c 147 "roswell":done
dispatch:--,name=top
proc_options:--:top
cmd_script_frontend:1
cmd_script
get_opt(program,0)=(null)
script_:argc=2 argv[0]=--
current=(null)
q 94784287059088 94784267747592 cmd-script.c 18 ""
s 94784287140336 cmd-script.c 23 "/usr/local/etc/roswell/setup.ros":done
set_opt(script)='"/usr/local/etc/roswell/setup.ros"'
s 94784287140432 cmd-script.c 26 ""/usr/local/etc/roswell/setup.ros"":done
q 94784287061296 94784287059280 util-list.c 178 "script"
s 94784287059280 util-list.c 147 "script":done
cmd_run_star:1:argv[0],script
s 94784287059280 util-dir.c 81 "ROSWELL_HOME":done
q 94784287061104 94784267749202 util-dir.c 83 ".roswell/"
get_opt(image,0)=(null)
get_opt(roswellenv,0)=roswell
get_opt(roswellenv,1)q 94784287061104 94784267750561 opt.c 114 "roswellenv"
s 94784287061104 opt.c 116 "roswellenv":done
=roswell
s 94784287061104 util-dir.c 81 "ROSWELL_HOME":done
q 94784287060912 94784267749202 util-dir.c 83 ".roswell/"
s 94784287140480 util-string.c 183 "/home/grey/.roswell/env/roswell/config":done
set_env_opt:/home/grey/.roswell/env/roswell/config
get_opt(lisp,1)q 94784287060496 94784267750573 opt.c 114 "lisp"
s 94784287060496 opt.c 116 "lisp":done
=(null)
get_opt(*lisp,0)=(null)
get_opt(default.lisp,0)=(null)
determin_impl:(null)
get_opt(default.lisp,1)q 94784287060496 94784267749532 opt.c 114 "default.lisp"
s 94784287060496 opt.c 116 "default_lisp":done
=(null)
q 94784287060496 94784267750425 cmd-run.c 83 "sbcl-bin"
s 94784287060496 cmd-run.c 93 "sbcl-bin":done
q 94784287060496 94784267750425 cmd-run.c 94 "sbcl-bin"
q 94784287060912 94784267753057 util.c 137 "/"
q 94784287060080 94784267748568 util.c 137 "tmp"
s 94784287059664 util-dir.c 81 "ROSWELL_HOME":done
q 94784287061488 94784267749202 util-dir.c 83 ".roswell/"
ensure_directories_exist:/home/grey/.roswell/tmp/
s 94784287140528 util-dir.c 44 "/home/grey/.roswell/tmp/":done
System:mkdir -p /home/grey/.roswell/tmp/
s 94784287140576 util-dir.c 40 "mkdir -p /home/grey/.roswell/tmp/":done
s 94784287140480 util.c 139 "/home/grey/.roswell/tmp/":done
q 94784287062672 94784267750339 util.c 140 "setup"
q 94784287062912 94784267748942 util.c 140 "tmp/lock.roswell."
s 94784287062480 util-dir.c 81 "ROSWELL_HOME":done
q 94784287062288 94784267749202 util-dir.c 83 ".roswell/"
lock setup exist status=0s 94784287132048 util.c 148 "/home/grey/.roswell/tmp/lock.roswell.setup":done
q 94784287062672 94784267753057 util.c 137 "/"
q 94784287062912 94784267748568 util.c 137 "tmp"
s 94784287061904 util-dir.c 81 "ROSWELL_HOME":done
q 94784287062288 94784267749202 util-dir.c 83 ".roswell/"
ensure_directories_exist:/home/grey/.roswell/tmp/
s 94784287140480 util.c 139 "/home/grey/.roswell/tmp/":done
q 94784287062672 94784267750339 util.c 140 "setup"
q 94784287062288 94784267748942 util.c 140 "tmp/lock.roswell."
s 94784287062912 util-dir.c 81 "ROSWELL_HOME":done
q 94784287061712 94784267749202 util-dir.c 83 ".roswell/"
lock!:setup
s 94784287132048 util.c 148 "/home/grey/.roswell/tmp/lock.roswell.setup":done
verbose-option:''
get_opt(sbcl-bin.version,0)=(null)
q 94784287062672 94784267747531 cmd-run.c 51 "config"
s 94784287062288 util-dir.c 81 "ROSWELL_HOME":done
q 94784287060912 94784267749202 util-dir.c 83 ".roswell/"
Installing sbcl-bin...
ros install sbcl-bin
System:ros install sbcl-bin
No SBCL version specified. Downloading platform-table.html to see the available versions...
[##########################################################################]100%
Installing sbcl-bin/1.5.5...
Downloading https://github.com/roswell/sbcl_bin/releases/download/1.5.5/sbcl-1.5.5-x86-64-linux-binary.tar.bz2
[##########################################################################]100%
Extracting sbcl-bin-1.5.5-x86-64-linux.tar.bz2 to /home/grey/.roswell/src/sbcl-1.5.5-x86-64-linux/
Building sbcl-bin/1.5.5... Done.
Install Script for sbcl-bin...
Installing Quicklisp...
up to date. stop
s 94784287061712 cmd-run.c 52 "ros install sbcl-bin":done
s 94784287140480 cmd-run.c 53 "/home/grey/.roswell/config":done
get_opt(sbcl-bin.version,0)=1.5.5
Making core for Roswell...
ros -N - -L sbcl-bin/1.5.5 setup roswell
System:ros -N - -L sbcl-bin/1.5.5 setup roswell
up to date. stop
s 94784287140688 cmd-run.c 61 "ros -N - -L sbcl-bin/1.5.5 setup roswell":done
q 94784287064096 94784267753057 util.c 137 "/"
q 94784287064288 94784267748568 util.c 137 "tmp"
s 94784287063904 util-dir.c 81 "ROSWELL_HOME":done
q 94784287063712 94784267749202 util-dir.c 83 ".roswell/"
ensure_directories_exist:/home/grey/.roswell/tmp/
s 94784287140576 util.c 139 "/home/grey/.roswell/tmp/":done
q 94784287064096 94784267750339 util.c 140 "setup"
q 94784287063712 94784267748942 util.c 140 "tmp/lock.roswell."
s 94784287064288 util-dir.c 81 "ROSWELL_HOME":done
q 94784287063328 94784267749202 util-dir.c 83 ".roswell/"
unlock!:setup
s 94784287140688 util.c 148 "/home/grey/.roswell/tmp/lock.roswell.setup":done
get_opt(sbcl-bin.version,0)=1.5.5
q 94784287066016 94784267753057 cmd-run.c 98 "/"
set_opt(impl)='sbcl-bin/1.5.5'
get_opt(quicklisp,0)=(null)
s 94784287140800 util-string.c 183 "/home/grey/.roswell/lisp/quicklisp/":done
set_opt(quicklisp)='/home/grey/.roswell/lisp/quicklisp/'
set_opt(argv0)='ros'
which cmd:command -v "ros"
q 94784287064864 94784267747592 util-system.c 9 ""
q 94784287066240 140733921801008 util-system.c 15 "/usr/local/bin/ros
"
which result:/usr/local/bin/ros

s 94784287067616 util.c 57 "/usr/local/bin/ros":done
s 94784287065056 util.c 57 "command -v "ros"":done
set_opt(wargv0)='/usr/local/bin/ros'
q 94784287067200 94784287060688 cmd-run.c 151 "/home/grey/.roswell/"
set_opt(homedir)='/home/grey/.roswell/'
set_opt(verbose)='3'
q 94784287067840 94784287022800 cmd-internal.c 157 "/usr/local/etc/roswell/"
q 94784287069184 94784287067840 cmd-run.c 153 "/usr/local/etc/roswell/"
set_opt(lispdir)='/usr/local/etc/roswell/'
q 94784287141184 94784287141136 cmd-run.c 154 "/usr/local/etc/roswell/patch/"
q 94784287141232 94784287141184 util-string.c 155 "/usr/local/etc/roswell/patch/"
set_opt(patchdir)='/usr/local/etc/roswell/patch/'
get_opt(asdf.version,0)=(null)
q 94784287068416 94784267747592 util-system.c 9 ""
q 94784287068224 140733921801008 util-system.c 15 "Linux
"
s 94784287068032 util.c 9 "Linux
":done
set_opt(uname)='linux'
q 94784287069376 94784267747592 util-system.c 9 ""
q 94784287070720 140733921801008 util-system.c 15 "x86_64
"
s 94784287070528 util.c 17 "x86_64
":done
set_opt(uname-m)='x86-64'
s 94784287060688 cmd-run.c 159 "/home/grey/.roswell/":done
get_opt(program,0)=(null)
q 94784287060688 94784267750663 cmd-run.c 164 "init.lisp"
s 94784287070336 util-dir.c 81 "ROSWELL_HOME":done
q 94784287070144 94784267749202 util-dir.c 83 ".roswell/"
q 94784287060688 94784267747592 cmd-run.c 170 ""
q 94784287069760 94784267747592 cmd-run.c 169 ""
q 94784287070144 94784267747592 cmd-run.c 168 ""
q 94784287141520 94784267750618 cmd-run.c 167 "(:eval"(ros:quicklisp)")"
get_opt(asdf.version,0)=(null)
q 94784287069952 94784267747592 cmd-run.c 166 ""
s 94784287141472 cmd-run.c 171 "/home/grey/.roswell/init.lisp":done
set_opt(program)='(:eval"(ros:quicklisp)")'
get_opt(impl,0)=sbcl-bin/1.5.5
s 94784287069952 util-dir.c 81 "ROSWELL_HOME":done
q 94784287069568 94784267749202 util-dir.c 83 ".roswell/"
q 94784287070944 94784267747592 util-system.c 9 ""
q 94784287072160 140733921800752 util-system.c 15 "x86_64
"
s 94784287071968 util.c 17 "x86_64
":done
q 94784287071968 94784267747592 util-system.c 9 ""
q 94784287070944 140733921800752 util-system.c 15 "Linux
"
s 94784287071744 util.c 9 "Linux
":done
get_opt(help,0)=(null)
get_opt(script,0)="/usr/local/etc/roswell/setup.ros"
get_opt(image,0)=(null)
get_opt(program,0)=(:eval"(ros:quicklisp)")
get_opt(dynamic-space-size,0)=(null)
get_opt(control-stack-size,0)=(null)
get_opt(without-roswell,0)=(null)
get_opt(enable-debugger,0)=(null)
s 94784287072160 cmd-run-sbcl.c 30 "x86-64":done
s 94784287070944 cmd-run-sbcl.c 30 "linux":done
s 94784287141776 cmd-run-sbcl.c 35 "/home/grey/.roswell/impls/x86-64/linux/sbcl-bin/1.5.5/lib/sbcl":done
q 94784287071968 94784267751242 cmd-run-sbcl.c 71 "--core"
q 94784287071360 94784267751285 cmd-run-sbcl.c 75 "--noinform"
get_opt(version,0)=(null)
q 94784287072544 94784267751355 cmd-run-sbcl.c 92 "--no-sysinit"
q 94784287073760 94784267751368 cmd-run-sbcl.c 93 "--no-userinit"
q 94784287073344 94784267751382 cmd-run-sbcl.c 95 "--disable-debugger"
get_opt(asdf.version,0)=(null)
s 94784287141936 cmd-run-sbcl.c 101 "/home/grey/.roswell/impls/x86-64/linux/sbcl-bin/1.5.5/fasl/init.lisp":done
q 94784287072960 94784267750663 cmd-run-sbcl.c 102 "init.lisp"
q 94784287072736 94784287022800 cmd-internal.c 157 "/usr/local/etc/roswell/"
s 94784287072736 util-string.c 183 "/usr/local/etc/roswell/":done
init.lisp=/usr/local/etc/roswell/init.lisp
q 94784287075552 94784267749555 cmd-run-sbcl.c 105 "--eval"
q 94784287075168 94784267751429 cmd-run-sbcl.c 106 ""))"
q 94784287142864 94784267751433 cmd-run-sbcl.c 106 "(progn #-ros.init(cl:load ""
s 94784287141616 cmd-run-sbcl.c 107 "impls/x86-64/linux/sbcl-bin/1.5.5":done
q 94784287074976 94784267749555 cmd-run-sbcl.c 109 "--eval"
q 94784287074560 94784267751430 cmd-run-sbcl.c 112 "))"
q 94784287141616 94784287141568 cmd-run-sbcl.c 110 "(:eval"(ros:quicklisp)")"
q 94784287074368 94784267751493 cmd-run-sbcl.c 110 "(ros:run '("

help=nil script="/usr/local/etc/roswell/setup.ros"
q 94784287141936 94784287141856 util-list.c 178 "/home/grey/.roswell/impls/x86-64/linux/sbcl-bin/1.5.5/bin/sbcl"
q 94784287074368 94784287071968 util-list.c 178 "--core"
q 94784287143136 94784287142768 util-list.c 178 "/home/grey/.roswell/impls/x86-64/linux/sbcl-bin/1.5.5/lib/sbcl/sbcl.core"
q 94784287072960 94784287071360 util-list.c 178 "--noinform"
q 94784287072736 94784287072544 util-list.c 178 "--no-sysinit"
q 94784287074176 94784287073760 util-list.c 178 "--no-userinit"
q 94784287075744 94784287073344 util-list.c 178 "--disable-debugger"
q 94784287077152 94784287075552 util-list.c 178 "--eval"
q 94784287143232 94784287141776 util-list.c 178 "(progn #-ros.init(cl:load "/usr/local/etc/roswell/init.lisp"))"
q 94784287076960 94784287074976 util-list.c 178 "--eval"
q 94784287143312 94784287143024 util-list.c 178 "(ros:run '((:eval"(ros:quicklisp)")(:script "/usr/local/etc/roswell/setup.ros")(:quit ())))"
s 94784287141856 util-list.c 147 "/home/grey/.roswell/impls/x86-64/linux/sbcl-bin/1.5.5/bin/sbcl":done
s 94784287071968 util-list.c 147 "--core":done
s 94784287142768 util-list.c 147 "/home/grey/.roswell/impls/x86-64/linux/sbcl-bin/1.5.5/lib/sbcl/sbcl.core":done
s 94784287071360 util-list.c 147 "--noinform":done
s 94784287072544 util-list.c 147 "--no-sysinit":done
s 94784287073760 util-list.c 147 "--no-userinit":done
s 94784287073344 util-list.c 147 "--disable-debugger":done
s 94784287075552 util-list.c 147 "--eval":done
s 94784287141776 util-list.c 147 "(progn #-ros.init(cl:load "/usr/local/etc/roswell/init.lisp"))":done
s 94784287074976 util-list.c 147 "--eval":done
s 94784287143024 util-list.c 147 "(ros:run '((:eval"(ros:quicklisp)")(:script "/usr/local/etc/roswell/setup.ros")(:quit ())))":done
s 94784287069760 cmd-run.c 193 "sbcl-bin":done
s 94784287070144 cmd-run.c 193 "1.5.5":done
q 94784287071968 94784267750661 cmd-run.c 205 ")"
q 94784287072160 94784267747592 opt.c 24 ""
q 94784287076768 94784267750660 opt.c 26 "")"
q 94784287076384 94784267747556 opt.c 26 """"
q 94784287076160 94784287061712 opt.c 26 "setup.time"
q 94784287075968 94784267747559 opt.c 26 "(""
q 94784287075552 94784267750660 opt.c 26 "")"
q 94784287075168 94784267747556 opt.c 26 """"
q 94784287074976 94784287060080 opt.c 26 "sbcl-bin.version"
q 94784287074752 94784267747559 opt.c 26 "(""
q 94784287075552 94784267750660 opt.c 26 "")"
q 94784287075168 94784267747556 opt.c 26 """"
q 94784287074976 94784287063136 opt.c 26 "default.lisp"
q 94784287074752 94784267747559 opt.c 26 "(""
q 94784287075552 94784267747592 opt.c 24 ""
q 94784287075360 94784267750660 opt.c 26 "")"
q 94784287075168 94784267747556 opt.c 26 """"
q 94784287074976 94784287060688 opt.c 26 "program"
q 94784287074752 94784267747559 opt.c 26 "(""
q 94784287075360 94784267750660 opt.c 26 "")"
q 94784287075168 94784267747556 opt.c 26 """"
q 94784287074752 94784287070528 opt.c 26 "uname-m"
q 94784287074560 94784267747559 opt.c 26 "(""
q 94784287075360 94784267750660 opt.c 26 "")"
q 94784287075168 94784267747556 opt.c 26 """"
q 94784287074752 94784287068032 opt.c 26 "uname"
q 94784287074560 94784267747559 opt.c 26 "(""
q 94784287075360 94784267750660 opt.c 26 "")"
q 94784287074976 94784267747556 opt.c 26 """"
q 94784287075168 94784287068608 opt.c 26 "patchdir"
q 94784287074752 94784267747559 opt.c 26 "(""
q 94784287075360 94784267750660 opt.c 26 "")"
q 94784287075168 94784267747556 opt.c 26 """"
q 94784287074752 94784287068992 opt.c 26 "lispdir"
q 94784287074560 94784267747559 opt.c 26 "(""
q 94784287075360 94784267750660 opt.c 26 "")"
q 94784287075168 94784267747556 opt.c 26 """"
q 94784287074752 94784287066432 opt.c 26 "verbose"
q 94784287074560 94784267747559 opt.c 26 "(""
q 94784287075360 94784267750660 opt.c 26 "")"
q 94784287075168 94784267747556 opt.c 26 """"
q 94784287074752 94784287067008 opt.c 26 "homedir"
q 94784287074560 94784267747559 opt.c 26 "(""
q 94784287075360 94784267750660 opt.c 26 "")"
q 94784287075168 94784267747556 opt.c 26 """"
q 94784287074752 94784287067616 opt.c 26 "wargv0"
q 94784287074560 94784267747559 opt.c 26 "(""
q 94784287075360 94784267750660 opt.c 26 "")"
q 94784287075168 94784267747556 opt.c 26 """"
q 94784287074752 94784287065440 opt.c 26 "argv0"
q 94784287074560 94784267747559 opt.c 26 "(""
q 94784287075360 94784267750660 opt.c 26 "")"
q 94784287074976 94784267747556 opt.c 26 """"
q 94784287075168 94784287065632 opt.c 26 "quicklisp"
q 94784287074752 94784267747559 opt.c 26 "(""
q 94784287075360 94784267750660 opt.c 26 "")"
q 94784287075168 94784267747556 opt.c 26 """"
q 94784287074752 94784287065824 opt.c 26 "impl"
q 94784287074560 94784267747559 opt.c 26 "(""
q 94784287075360 94784267750660 opt.c 26 "")"
q 94784287074976 94784267747556 opt.c 26 """"
q 94784287075168 94784287059472 opt.c 26 "script"
q 94784287074752 94784267747559 opt.c 26 "(""
q 94784287075360 94784267750660 opt.c 26 "")"
q 94784287075168 94784267747556 opt.c 26 """"
q 94784287074752 94784287058256 opt.c 26 "roswellenv"
q 94784287074560 94784267747559 opt.c 26 "(""
q 94784287075360 94784267751503 cmd-run.c 205 "("
get_opt(wrap,1)q 94784287075360 94784267748265 opt.c 114 "wrap"
s 94784287075360 opt.c 116 "wrap":done
=(null)
args /home/grey/.roswell/impls/x86-64/linux/sbcl-bin/1.5.5/bin/sbcl --core /home/grey/.roswell/impls/x86-64/linux/sbcl-bin/1.5.5/lib/sbcl/sbcl.core --noinform --no-sysinit --no-userinit --disable-debugger --eval (progn #-ros.init(cl:load "/usr/local/etc/roswell/init.lisp")) --eval (ros:run '((:eval"(ros:quicklisp)")(:script "/usr/local/etc/roswell/setup.ros")(:quit ())))
ROS_OPTS (("program""(:eval\"(ros:quicklisp)\")")("uname-m""x86-64")("uname""linux")("patchdir""/usr/local/etc/roswell/patch/")("lispdir""/usr/local/etc/roswell/")("verbose""3")("homedir""/home/grey/.roswell/")("wargv0""/usr/local/bin/ros")("argv0""ros")("quicklisp""/home/grey/.roswell/lisp/quicklisp/")("impl""sbcl-bin/1.5.5")("script""\"/usr/local/etc/roswell/setup.ros\"")("roswellenv""roswell")("setup.time""3775285962")("sbcl-bin.version""wï4V")("default.lisp""sbcl-bin"))
Unhandled TYPE-ERROR in thread #<SB-THREAD:THREAD "main thread" RUNNING
                                  {10005084C3}>:
  The value
    NIL
  is not of type
    (OR (VECTOR CHARACTER) (VECTOR NIL) BASE-STRING PATHNAME SYNONYM-STREAM
        FILE-STREAM)

  when binding SB-IMPL::DEFAULTS

Backtrace for: #<SB-THREAD:THREAD "main thread" RUNNING {10005084C3}>
0: (MERGE-PATHNAMES "setup.lisp" NIL #<FUNCTION EQUAL>) [external]
1: (MAIN "roswell")
2: (SB-INT:SIMPLE-EVAL-IN-LEXENV (APPLY (QUOTE MAIN) ROSWELL:*ARGV*) #<NULL-LEXENV>)
3: (SB-INT:SIMPLE-EVAL-IN-LEXENV (ROSWELL:QUIT (APPLY (QUOTE MAIN) ROSWELL:*ARGV*)) #<NULL-LEXENV>)
4: (SB-EXT:EVAL-TLF (ROSWELL:QUIT (APPLY (QUOTE MAIN) ROSWELL:*ARGV*)) NIL NIL)
5: ((LABELS SB-FASL::EVAL-FORM :IN SB-INT:LOAD-AS-SOURCE) (ROSWELL:QUIT (APPLY (QUOTE MAIN) ROSWELL:*ARGV*)) NIL)
6: (SB-INT:LOAD-AS-SOURCE #<CONCATENATED-STREAM :STREAMS NIL {1002CE0B43}> :VERBOSE NIL :PRINT NIL :CONTEXT "loading")
7: ((FLET SB-FASL::THUNK :IN LOAD))
8: (SB-FASL::CALL-WITH-LOAD-BINDINGS #<CLOSURE (FLET SB-FASL::THUNK :IN LOAD) {7FEB0C04F53B}> #<CONCATENATED-STREAM :STREAMS NIL {1002CE0B43}>)
9: ((FLET SB-FASL::LOAD-STREAM :IN LOAD) #<CONCATENATED-STREAM :STREAMS NIL {1002CE0B43}> NIL)
10: (LOAD #<CONCATENATED-STREAM :STREAMS NIL {1002CE0B43}> :VERBOSE NIL :PRINT NIL :IF-DOES-NOT-EXIST T :EXTERNAL-FORMAT :DEFAULT)
11: ((FLET ROSWELL::BODY :IN ROSWELL:SCRIPT) #<SB-SYS:FD-STREAM for "file /usr/local/etc/roswell/setup.ros" {1002CDB873}>)
12: (ROSWELL:SCRIPT "/usr/local/etc/roswell/setup.ros")
13: (ROSWELL:RUN ((:EVAL "(ros:quicklisp)") (:SCRIPT "/usr/local/etc/roswell/setup.ros") (:QUIT NIL)))
14: (SB-INT:SIMPLE-EVAL-IN-LEXENV (ROSWELL:RUN (QUOTE ((:EVAL "(ros:quicklisp)") (:SCRIPT "/usr/local/etc/roswell/setup.ros") (:QUIT NIL)))) #<NULL-LEXENV>)
15: (EVAL (ROSWELL:RUN (QUOTE ((:EVAL "(ros:quicklisp)") (:SCRIPT "/usr/local/etc/roswell/setup.ros") (:QUIT NIL)))))
16: (SB-IMPL::PROCESS-EVAL/LOAD-OPTIONS ((:EVAL . "(progn #-ros.init(cl:load \"/usr/local/etc/roswell/init.lisp\"))") (:EVAL . "(ros:run '((:eval\"(ros:quicklisp)\")(:script \"/usr/local/etc/roswell/setup.ros\")(:quit ())))")))
17: (SB-IMPL::TOPLEVEL-INIT)
18: ((FLET SB-UNIX::BODY :IN SB-EXT:SAVE-LISP-AND-DIE))
19: ((FLET "WITHOUT-INTERRUPTS-BODY-7" :IN SB-EXT:SAVE-LISP-AND-DIE))
20: ((LABELS SB-IMPL::RESTART-LISP :IN SB-EXT:SAVE-LISP-AND-DIE))

unhandled condition in --disable-debugger mode, quitting
```

**Comment by snmsts** [comment]
I don't have idea how to reproduce.

**Comment by mister-walter** [comment]
Ran into this again, this time in a Docker image using the debian-slim base image.

I've created a branch named `drew/roswell_bug` in the [relevant repo](https://github.com/mister-walter/acl2-docker/tree/drew/roswell_bug), you need to have the `jq` binary installed but other than that you can clone that repo and run `make build` to see the error occur.

For what it's worth, the digest for the `docker-slim` image is `sha256:a8787fc446fca239c3475b0e7890c202fb8abfee343082418dabdce060cf4f81`; it's just the latest version. Roswell was working well on a version of the `docker-slim` image from about 4 months ago.

**Comment by mister-walter** [comment]
@tcw Installing `libev-dev` did not work but thank you for the suggestion!

**Comment by vrx** [comment]
i get the same error

uname -a
Linux yoga 5.0.0-32-generic #34~18.04.2-Ubuntu SMP Thu Oct 10 10:36:02 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux

**Comment by vrx** [comment]
well, after a reboot, the error disappeared...
ros works fine for me on ubuntu 18.04
thanks !

**Comment by windymelt** [comment]
Hi, I hit same problem in Docker (`debian:buster`). Finally I solved the problem using `debian:stretch-slim` instead of `debian:buster`.

I suspecting version difference of `glibc` between buster and stretch.

**Comment by MackEdweise** [comment]
Ran into this as well. Solution for me was as simple as ending my terminal session, starting a new one, and re-running `ros setup`. No issues the second time and sbcl is working fine :)

**Comment by snmsts** [comment]
@windymelt could you share docker file?

**Comment by duncan-bayne** [comment]
@windymelt Thank you!  I was experiencing this exact problem using the Docker-based CI pipeline in GitLab.  Switching to `debian:stretch-slim` fixed the issue.

**Comment by windymelt** [comment]
@snmsts 

Problem reproduction:

```dockerfile
FROM debian:buster

# almost same as https://github.com/roswell/roswell/wiki/Installation#building-from-source

RUN apt-get -y update && apt-get -y install git build-essential automake libcurl4-openssl-dev
RUN git clone -b release https://github.com/roswell/roswell.git && cd roswell && sh bootstrap && ./configure && make && make install
RUN ros setup
```

You should erase problem using `FROM debian:stretch-slim`.

**Comment by snmsts** [comment]
@windymelt confirmed thank you.

hmmm.

**Comment by snmsts** [comment]
fixed with 20.01.14.104

**Comment by phlummox** [comment]
> fixed with 20.01.14.104

Are you able to say what the cause of the bug was, and roughly what the fix involved? Since Ubuntu 18.04 still seems to be having issues.
